packages/backend/src/helpers/flow-templates.ts

```diff
@@ -1,83 +1,110 @@
+import type { IApp } from '@plumber/types'
+
+import apps from '@/apps'
+import { TEMPLATES } from '@/db/storage'
 import type {
-  AppEventKeyPair,
-  CreateTemplatedFlowInput,
   FlowConfig,
+  TemplateStep,
 } from '@/graphql/__generated__/types.generated'
 import Flow from '@/models/flow'
 import User from '@/models/user'
 
 import logger from './logger'
 
-export const DEFAULT_FLOW_TEMPLATE: CreateTemplatedFlowInput = {
-  flowName: 'Send notifications',
-  trigger: {
-    appKey: 'formsg',
-    eventKey: 'newSubmission',
-  },
-  actions: [
-    {
-      appKey: 'postman',
-      eventKey: 'sendTransactionalEmail',
-    },
-  ],
-  demoVideoId: 'formsg-postman',
+function validateAppAndEventKey(step: TemplateStep, templateName: string) {
+  let app: IApp
+  const { position, appKey, eventKey } = step
+  if (appKey) {
+    app = apps[appKey]
+    if (!app) {
+      throw new Error(
+        `Invalid app key for ${templateName} template at step ${position}`,
+      )
+    }
+  }
+
+  if (eventKey) {
+    const event = app?.triggers
+      ? app?.triggers.find((trigger) => trigger.key === eventKey)
+      : app?.actions.find((action) => action.key === eventKey)
+    if (!event) {
+      throw new Error(
+        `Invalid event key for ${templateName} template at step ${position}`,
+      )
+    }
+  }
 }
 
-async function createFlowFromTemplate(
-  flowName: string,
-  trigger: AppEventKeyPair,
-  actions: AppEventKeyPair[],
+export async function createFlowFromTemplate(
+  templateId: string,
   user: User,
   isAutoCreated: boolean,
```

Comment:
can take this time to use named params

Reply:
isAutoCreated will be removed later right

Reply:
yessir, it should only have 2 parameters, so i don't think i would need named params?